plugins/extractit/icon_extractit/util/extractor.py

```diff
@@ -126,8 +125,32 @@ def clear_emails(matches: list) -> list:
     return new_matches
 
 
-def parse_time(dates: list) -> list:
+def define_linux_date_time_format(date_format: str) -> str:
+    return DateFormatStrings.human_to_linux_mapping.get(date_format)
+
+
+def define_date_time_regex(date_format: str) -> str:
+    return Regex.HumanToRegexPatterns.get(date_format)
+
+
+def parse_time(dates: list, date_format: str) -> list:
+    linux_date_time_format = define_linux_date_time_format(date_format)
+    # Regex pattern used for identification of separators to handle date time formats in ISO format and all possible
+    # cases of mixed separator format that can be converted to a datetime object
+    date_value_separators = regex.findall(r"[^(%b)(%d)(%m)(%y)(%h)(%s)(%D)(%M)(%Y)(%H)(%S)]",linux_date_time_format)
+
     for date in enumerate(dates):
-        date_time_obj = datetime.strptime(date[1], "%d/%m/%Y")
-        dates[date[0]] = date_time_obj.strftime("%Y-%m-%dT%H:%M:%SZ")
+        date_string_list = [None] * (len(date[1]) + len(date_value_separators))
+        date_string_list[::2] = date[1];
+        date_string_list[1::2] = date_value_separators
+        date_string = ''.join(date_string_list)
+        try:
```

Comment:
Unsure if best to raise error on incorrect date format or ignore

Reply:
@ablakley-r7 does this correspond to the date format inputs that you added to the actions? If so - since the input is an enum, there can never be an "unknown" date format provided.